app/workers/document_list_export_worker.rb

```diff
@@ -6,14 +6,28 @@ def perform(filter_options, user_id)
     Tempfile.open("document_list_export_worker") do |file|
       file.unlink
       csv = generate_csv(filter, file)
-      send_mail(csv, user, filter)
+      public_url = upload_file(csv, filter_options[:type])
+      send_mail(public_url, user, filter)
     end
   end
 
 private
 
-  def send_mail(csv, user, filter)
-    Notifications.document_list(csv, user.email, filter.page_title).deliver_now
+  def send_mail(csv_filename, user, filter)
+    Notifications.document_list(csv_filename, user.email, filter.page_title).deliver_now
+  end
+
+  def upload_file(csv, type)
+    export_id = SecureRandom.uuid
+    document_type_slug = if type.present?
+                           type.downcase.gsub(%r{[^a-z0-9]+}, "_")
+                         else
+                           "documents"
+                         end
+
+    filename = "document_list_#{document_type_slug}_#{export_id}.csv"
```

Comment:
It seems a shame / risky to duplicate this logic here and in the `show` method of the controller - could it be pulled out?

Reply:
I'm not sure where the best place to move it to would be: neither the controller nor the worker seem right, and there isn't a model for this — or would creating one simply for that make sense?

Reply:
Perhaps in `DocumentListExportPresenter`?